client/gutenberg/extensions/tiled-gallery/index.js

```diff
@@ -118,6 +118,26 @@ export const settings = {
 	title: __( 'Tiled Gallery' ),
 	transforms: {
 		from: [
+			{
+				type: 'block',
+				isMultiBlock: true,
+				blocks: [ 'core/image' ],
+				transform: attributes => {
+					const validImages = filter( attributes, ( { id, url } ) => id && url );
+					if ( validImages.length > 0 ) {
+						return createBlock( `jetpack/${ name }`, {
+							images: validImages.map( ( { id, url, alt, caption } ) => ( {
+								id,
+								url,
+								alt,
+								caption,
+							} ) ),
+							ids: validImages.map( ( { id } ) => id ),
+						} );
+					}
+					return createBlock( `jetpack/${ name }` );
```

Comment:
Good catch on the 0 valid images, although rather than creating an empty gallery, what about using `isMatch` to make the transform impossible?

https://wordpress.org/gutenberg/handbook/designers-developers/developers/block-api/block-registration/#transforms-optional

Reply:
Good idea. I used it for this `core/image` transform in https://github.com/Automattic/wp-calypso/pull/30301/commits/a79cc5aa4f75866404e6490f72035a7db08d45c3

I looked into using it with other transforms as well but they're a bit different:
- they're not `isMultiBlock` transforms
- I'd expect to be able to transform empty core gallery to tiled gallery etc